test/nodes/test_link_content_retriever.py

```diff
@@ -0,0 +1,350 @@
+from unittest.mock import Mock, patch
+import logging
+import pytest
+import requests
+from requests import Response
+
+from haystack import Document
+from haystack.nodes import LinkContentRetriever
+
+
+@pytest.fixture
+def mocked_requests():
+    with patch("haystack.nodes.retriever.link_content.requests") as mock_requests:
+        mock_response = Mock()
+        mock_requests.get.return_value = mock_response
+        mock_response.status_code = 200
+        mock_response.text = "Sample content from webpage"
+        yield mock_requests
+
+
+@pytest.fixture
+def mocked_article_extractor():
+    with patch("boilerpy3.extractors.ArticleExtractor.get_content", return_value="Sample content from webpage"):
+        yield
+
+
+@pytest.mark.unit
+def test_init():
+    """
+    Checks the initialization of the LinkContentRetriever without a preprocessor.
+    """
+    r = LinkContentRetriever()
+
+    assert r.processor is None
+    assert isinstance(r.handlers, dict)
+    assert "html" in r.handlers
+
+
+@pytest.mark.unit
+def test_init_with_preprocessor():
+    """
+    Checks the initialization of the LinkContentRetriever with a preprocessor.
+    """
+    pre_processor_mock = Mock()
+    r = LinkContentRetriever(processor=pre_processor_mock)
+    assert r.processor == pre_processor_mock
+    assert isinstance(r.handlers, dict)
+    assert "html" in r.handlers
+
+
+@pytest.mark.unit
+def test_call(mocked_requests, mocked_article_extractor):
```

Comment:
```suggestion
def test_fetch(mocked_requests, mocked_article_extractor):
```
This and the following tests could be renamed with `fetch` instead of `call`.